### PR TITLE
Update MetalLB env variables to fit the latest changes in dev-scripts

### PR DESCRIPTION
**Pull request by sabinaaledort**
Following these changes:
https://github.com/openshift-metal3/dev-scripts/pull/1302
https://github.com/openshift-metal3/dev-scripts/pull/1305

### CODE REVIEW

**Comment by sabinaaledort** [comment]
/hold
should be merged after https://github.com/openshift-metal3/dev-scripts/pull/1305

**Comment by sabinaaledort** [comment]
METALLB_OPERATOR_COMMIT should be updated according to https://github.com/openshift/metallb-operator/pull/43

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: Why hard-coding to a commit SHA?
Can't METALLB_OPERATOR_COMMIT be set to "main" so that it keeps up with latest main branch?

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:75]: same as above

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.10.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-release-4.10.yaml:63]: Same as above. In this case, can't it be set to "release-4.10" to follow the OCP 4.10 release branch?

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.10.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-release-4.10.yaml:76]: same as above

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml:62]: Same as above. This is even worse for the time being, no? It means we would be installing a newer version than the one shipped in OCP 4.9.

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml
Comment (cgoncalves) [inline ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml:75]: Same as above.

**Review by cgoncalves — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: if we do a change in the operator that's not reflected in metallb we might break the CI

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml
Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml:62]: here im still wondering if we are going to backport the FRR stuff to 4.9 both in metallb and in the operator

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (cgoncalves) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: I don't follow. Could you please share an example?

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml
Comment (cgoncalves) [reply ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml:62]: We will not.

Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml:62]: so i will set it to a 4.9 commit

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: so this operator commit id is for deploying metallb through the operator is d/s metallb CI right? passed to the devscripts here
https://github.com/openshift-metal3/dev-scripts/blob/918da5cba37ec38b47e67dbe08d53297e8dcb7ea/metallb/deploy_operator.sh#L7
if we do a big change in the operator it will immediately affect the deployment of the operator for metallb CI, so it might break all our PRs in metallb repo

Comment (cgoncalves) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: So your example is of a potential disruptive MetalLB Operator PR that breaks MetalLB deployments, correct?
If such a big change in the operator breaks MetalLB CI, it should be caught by CI on the MetalLB Operator PR itself before it is merged in the first place and impacts the MetalLB CI. Am I missing something?

File: ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml
Comment (cgoncalves) [reply ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml]: I was more suggesting setting it to "release-4.9" than a specific commit.

Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-release-4.9.yaml]: i know, let's decide first if we want a branch or commit.

File: ci-operator/config/openshift/metallb/openshift-metallb-main.yaml
Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: @fedepaol what do you think?

Comment (fedepaol) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: The way it works is, a new PR on the operator (or on the frr image, or on metallb) builds the images, and uses whatever it is in terms of manifests in the operator to deploy and test them. This is not optimal, but will cover the 90% of our cases.
@sabinaaledort 's concern is legit, and we won't able to block a bad PR from the operator. On the other hand, having everything broken will force us to fix it :-).

So I'd choose pinning to a branch.

Comment (sabinaaledort) [reply ci-operator/config/openshift/metallb/openshift-metallb-main.yaml:62]: so branch it is :)

**Comment by sabinaaledort** [comment]
failing to deploying the operator as the operator d/s repo wasn't updated yet. anyway it will probably fail after the merge as well because we don't have the updated metallb E2E tests in the d/s metallb repo yet.

**Comment by sabinaaledort** [comment]
/retest
openshift/metallb-operator#43 was merged

**Comment by sabinaaledort** [comment]
metallb-operator and metallb were deployed successfully 
```
2021-11-24 09:24:07 ENABLE_OPERATOR_WEBHOOK=true hack/configure_operator_webhook.sh
2021-11-24 09:24:07 cd config/manager && /root/go/bin/kustomize edit set image controller=virthost.ostest.test.metalkube.org:5000/metallb-operator:ci
2021-11-24 09:24:07 cd ../metallb-operator-deploy && /root/go/bin/kustomize edit set namespace metallb-system
2021-11-24 09:24:07 cd config/metallb_rbac && /root/go/bin/kustomize edit set namespace metallb-system
2021-11-24 09:24:07 /root/go/bin/kustomize build ../metallb-operator-deploy | kubectl apply -f -
2021-11-24 09:24:08 namespace/metallb-system created
2021-11-24 09:24:08 customresourcedefinition.apiextensions.k8s.io/addresspools.metallb.io created
2021-11-24 09:24:08 customresourcedefinition.apiextensions.k8s.io/bfdprofiles.metallb.io created
2021-11-24 09:24:08 customresourcedefinition.apiextensions.k8s.io/bgppeers.metallb.io created
2021-11-24 09:24:08 customresourcedefinition.apiextensions.k8s.io/metallbs.metallb.io created
2021-11-24 09:24:08 role.rbac.authorization.k8s.io/metallb-operator-leader-election-role created
2021-11-24 09:24:08 role.rbac.authorization.k8s.io/metallb-operator-manager-role created
2021-11-24 09:24:08 clusterrole.rbac.authorization.k8s.io/metallb-operator-manager-role created
2021-11-24 09:24:08 rolebinding.rbac.authorization.k8s.io/metallb-operator-leader-election-rolebinding created
2021-11-24 09:24:08 rolebinding.rbac.authorization.k8s.io/metallb-operator-manager-rolebinding created
2021-11-24 09:24:09 clusterrolebinding.rbac.authorization.k8s.io/metallb-operator-manager-rolebinding created
2021-11-24 09:24:09 service/webhook-service created
2021-11-24 09:24:09 deployment.apps/metallb-operator-controller-manager created
2021-11-24 09:24:09 validatingwebhookconfiguration.admissionregistration.k8s.io/validating-webhook-configuration created
2021-11-24 09:24:09 /root/go/bin/kustomize build config/metallb_rbac | kubectl apply -f -
2021-11-24 09:24:09 serviceaccount/controller created
2021-11-24 09:24:09 serviceaccount/speaker created
2021-11-24 09:24:09 role.rbac.authorization.k8s.io/config-watcher created
2021-11-24 09:24:09 role.rbac.authorization.k8s.io/controller created
2021-11-24 09:24:09 role.rbac.authorization.k8s.io/pod-lister created
2021-11-24 09:24:09 role.rbac.authorization.k8s.io/speaker created
2021-11-24 09:24:09 clusterrole.rbac.authorization.k8s.io/metallb-system:controller created
2021-11-24 09:24:09 clusterrole.rbac.authorization.k8s.io/metallb-system:speaker created
2021-11-24 09:24:09 rolebinding.rbac.authorization.k8s.io/config-watcher created
2021-11-24 09:24:09 rolebinding.rbac.authorization.k8s.io/controller created
2021-11-24 09:24:09 rolebinding.rbac.authorization.k8s.io/pod-lister created
2021-11-24 09:24:09 rolebinding.rbac.authorization.k8s.io/speaker created
2021-11-24 09:24:09 clusterrolebinding.rbac.authorization.k8s.io/metallb-system:controller created
2021-11-24 09:24:09 clusterrolebinding.rbac.authorization.k8s.io/metallb-system:speaker created
2021-11-24 09:24:09 make[1]: Leaving directory '/root/dev-scripts/metallb/metallb-operator'
2021-11-24 09:24:09 +(./configure_metallb.sh:8): main(): oc apply -f /root/dev-scripts/metallb/metallb.yaml
2021-11-24 09:24:10 metallb.metallb.io/metallb created
2021-11-24 09:24:10 +(./configure_metallb.sh:9): main(): oc adm policy add-scc-to-user privileged -n metallb-system -z speaker
2021-11-24 09:24:10 clusterrole.rbac.authorization.k8s.io/system:openshift:scc:privileged added: "speaker"
```
the pr is still failing as we don't have the updated tests in the d/s metallb repo
`No idea what '--ipv4-service-range' is!`
the following pr will add it:
https://github.com/openshift/metallb/pull/13

**Pull request merged**
